automation/check-patch.k8s-1.14.2-kind-sriov.sh

```diff
@@ -0,0 +1,242 @@
+#
+#############################################################
+
+
+#############################################################
+# This is based on https://github.com/SchSeba/kubevirt-docker
+#############################################################
+
+export NO_PROXY="localhost,127.0.0.1,172.17.0.2"
+
+export WORKSPACE="${WORKSPACE:-$PWD}"
+readonly ARTIFACTS_PATH="${ARTIFACTS-$WORKSPACE/exported-artifacts}"
+
+CLUSTER_NAME=sriov-ci
+CLUSTER_CONTROL_PLANE=${CLUSTER_NAME}-control-plane
+CONTAINER_REGISTRY_HOST="localhost:5000"
+
+CLUSTER_CMD="docker exec -it -d ${CLUSTER_CONTROL_PLANE}"
+
+KUBEVIRT_PATH=`pwd`
+CLUSTER_DIR="cluster-up/cluster/k8s-1.14.2-kind-sriov"
+MANIFESTS_DIR="${CLUSTER_DIR}/manifests"
+
+
+function wait_containers_ready {
+    echo "Waiting for all containers to become ready ..."
+    kubectl wait --for=condition=Ready pod --all -n kube-system --timeout 12m
+}
+
+function wait_kubevirt_up {
+    echo "Waiting for kubevirt to be ready ..."
+    kubectl wait --for=condition=Ready pod --all -n kubevirt --timeout 12m
+    kubectl wait --for=condition=Ready pod --all -n cdi --timeout 12m
+}
+
+function wait_kind_up {
+    while [ -z "$(docker exec --privileged ${CLUSTER_CONTROL_PLANE} kubectl --kubeconfig=/etc/kubernetes/admin.conf get nodes --selector=node-role.kubernetes.io/master -o=jsonpath='{.items..status.conditions[-1:].status}' | grep True)" ]; do
+        echo "Waiting for kind to be ready ..."        
+	    sleep 10
+    done
+}
+
+function collect_artifacts {
+    set +e
+    mkdir -p "$ARTIFACTS_PATH/containers"
+    for c in $(docker ps -a | tail -n +2 | awk '{print$1}'); do
+        docker logs $c 2>&1 > $ARTIFACTS_PATH/containers/$c.log || true
+    done
+    kind export logs ${ARTIFACTS_PATH} --name=${CLUSTER_NAME} || true
+}
+
+function finish {
+    collect_artifacts
+    echo "Printing all pods"
+    kubectl get pods --all-namespaces -o wide
+    kind delete cluster --name=${CLUSTER_NAME}
+}
+
+function enable_vfio {
+    mount -o remount,rw /sys    #need this to move devices to vfio drivers
+    counter=0
+    for file in $(find /sys/devices/ -name *sriov_totalvfs*); do
+        pfroot=$(dirname $file)
+
+        # enable all enabled VFs. If it fails means that sysfs is not supported on that device and we pass
+        cat $file > $pfroot/sriov_numvfs || continue 
+
+        # bind all VFs with vfio
+        for virtfn in $(ls -d $pfroot/virtfn*); do
+            pciid=$(basename $(readlink $virtfn))
+            if [ -e $virtfn/driver/unbind ]; then
+                echo $pciid > $virtfn/driver/unbind
+            fi
+            echo $(lspci -n -s $pciid | sed 's/:/ /g' | awk '{print $4 " " $5}') > /sys/bus/pci/drivers/vfio-pci/new_id
+            counter=$((counter+1))
+        done
+    done
+}
+
+enable_vfio
+
+# Create the cluster...
+wget https://github.com/kubernetes-sigs/kind/releases/download/v0.3.0/kind-linux-amd64 -O /usr/local/bin/kind
```

Comment:
0.4.0 was released past week, there are several improvements and new features, however, we made same changes that can break compatibility :sw

Reply:
Yes, I saw it but I felt it was too fresh :smile: